src/compiler/pre-transform/index.ts

```diff
@@ -0,0 +1,221 @@
+import pkg from '@storybook/addon-svelte-csf/package.json' with { type: 'json' };
+import type { ImportDeclaration, VariableDeclaration } from 'estree';
+import type { Comment, Fragment, Root, Script, SvelteNode } from 'svelte/compiler';
+
+import { transformComponentMetaToDefineMeta } from '#compiler/pre-transform/codemods/component-meta-to-define-meta';
+import { transformExportMetaToDefineMeta } from '#compiler/pre-transform/codemods/export-const-to-define-meta';
+import { transformImportDeclaration } from '#compiler/pre-transform/codemods/import-declaration';
+import { transformLegacyStory } from '#compiler/pre-transform/codemods/legacy-story';
+import { transformTemplateToSnippet } from '#compiler/pre-transform/codemods/template-to-snippet';
+import { MissingModuleTagError } from '#utils/error/parser/extract/svelte';
+
+interface Params {
+  ast: Root;
+  filename?: string;
+}
+
+export async function codemodLegacyNodes(params: Params): Promise<Root> {
+  const { walk } = await import('zimmerframe');
+
+  const { ast, filename } = params;
+
+  interface State {
+    componentIdentifierName: ComponentIdentifierName;
+    componentMetaHtmlComment?: Comment;
+    defineMetaFromMeta?: VariableDeclaration;
+  }
+  const state: State = {
+    componentIdentifierName: {},
+  };
+  let transformedAst = walk(ast as SvelteNode | Script, state, {
+    _(_node, context) {
+      const { next, state } = context;
+
+      next(state);
+    },
+
+    Root(node, context) {
+      const { fragment, module, ...rest } = node;
+      const { state, visit } = context;
+
+      if (!module) {
+        throw new MissingModuleTagError(filename);
+      }
+
+      const transformedModule = visit(module, state) as Script;
+      const transformedFragment = visit(fragment, state) as Fragment;
+
+      return {
+        ...rest,
+        fragment: transformedFragment,
+        module: transformedModule,
+      };
+    },
+
+    Script(node, context) {
+      const { next, state } = context;
+
+      if (node.context === 'module') {
+        next(state);
+      }
+    },
+
+    Program(_node, context) {
+      const { next, state } = context;
+
+      next(state);
+    },
```

Comment:
This instructs the AST walker - `zimmerframe` to go "deeper" in the tree. I'm not sure if this can be omitted, but I can test.

`Root` -> `module` _(`Script`)_ -> `Program` -> `nodes` _(declarations)_.